src/client-lib/api.js

```diff
@@ -0,0 +1,44 @@
+import fetch from 'isomorphic-fetch';
+
+const API_PREFIX = '/api/v0';
+
+function makeRequest(uri, options) {
+    let res;
+    return fetch(uri, options)
+    .then(response => {
+        res = response;
+        return res.json();
+    })
+    .then(body => {
+        if (res.status >= 200 && res.status < 300) {
```

Comment:
I refactored it to return [res, body] so we don't have the awkward dangling res var, I hope you like it.